token-metadata-creator/app/Config.hs

```diff
@@ -196,8 +196,35 @@ wellKnownOption
     => OA.Mod OA.OptionFields p
     -> OA.Parser p
 wellKnownOption =
-    OA.option wellKnownReader
+  OA.option (OA.eitherReader (\arg ->
+                               let
+                                 r1 = asJSONValue arg
+                               in
+                                 if isPotentiallyString r1
+                                 then asJSONString arg
+                                 else r1
+                             ))
   where
-    wellKnownReader :: OA.ReadM p
-    wellKnownReader = OA.eitherReader $
-        Aeson.parseEither parseWellKnown . Aeson.toJSON
+    -- Presume we got a value (i.e. 3 or []).
+    asJSONValue :: String -> Either String p
+    asJSONValue = (Aeson.parseEither parseWellKnown =<<) . Aeson.eitherDecodeStrict' . BC8.pack
```

Comment:
```suggestion
    asJSONValue :: ByteString -> Either String p
    asJSONValue = Aeson.parseEither parseWellKnown <=< Aeson.eitherDecodeStrict'
```

Reply:
Hmm, that solution just moves the decoding error to the `UTF-8` locale. The `C` locale works, but `UTF-8` throws the same error. I'll look into the character encoding a bit more, I imagine I'm going to have to normalize the input stream.

Reply:
Which is exactly what you were saying... Sorry.

Reply:
No prob!